libs/components/src/lib/side-sheet/vl-side-sheet.component.ts

```diff
@@ -48,25 +52,38 @@ export class VlSideSheet extends BaseElementOfType(HTMLElement) {
         ${styles}
       </style>
       <div>
-        <button is="vl-button" type="button" class="vl-side-sheet__toggle">
-          <span is="vl-icon" data-vl-icon="nav-left"></span>
-          <span id="vl-side-sheet-toggle-text" is="vl-text" data-vl-visually-hidden>Zijpaneel</span>
-        </button>
-        <div id="vl-side-sheet-backdrop"></div>
-        <div id="vl-side-sheet">
-          <section is="vl-region">
-            <div is="vl-layout">
-              <slot></slot>
-            </div>
-          </section>
-        </div>
+          <vl-toggle-button
+              data-vl-icon="nav-left"
+              data-vl-icon-placement="before"
+              class="vl-side-sheet__toggle"
+          >
+            <span id="vl-side-sheet-toggle-text" is="vl-text"></span>
+          </vl-toggle-button>
+          <div id="vl-side-sheet-backdrop"></div>
+          <div id="vl-side-sheet">
+            <section is="vl-region">
+              <div is="vl-layout">
+                <slot></slot>
+              </div>
+            </section>
+          </div>
       </div>
     `);
     }
 
     connectedCallback() {
         this._toggle = () => this.toggle();
         this._toggleButton.addEventListener('click', this._toggle);
+        this._toggleButton.active = false;
+        this.updateToggleText(this.toggleText);
+        if (this.customIcon) {
+            this._toggleButton.setAttribute('data-vl-icon', this.customIcon);
```

Comment:
Kan je dit in een changedCallback uitvoeren? Dan werkt dat ook ineens reactief, ik zie dat je dit al doet voor de iconPlacement.